packages/data-context/test/unit/sources/migration/codegen.spec.ts

```diff
@@ -25,7 +25,7 @@ describe('cypress.config.js generation', () => {
 
     const generatedConfig = await createConfigString(config, {
       hasE2ESpec: true,
-      hasComponentTesting: false,
+      hasComponentTesting: true,
```

Comment:
Why was this change necessary? 
It feels like something that should generate a different string than before.

We also stopped testing `hasComponenttesting: false` altogether

Reply:
I'm curious as well why this changed.

Reply:
`should create only an e2e entry when no component specs are detected`, added below, asserts that `component` is left out when `hasComponenttesting: false`, so that case is still covered.

Previously we didn't have any tests covering both CT and e2e testing both being true. With this switched to true, we're asserting

"this global option remains global and doesn't end up in any testing-type specific config"
rather than
"this global option remains global and doesn't end up in the e2e config"

I think the default should be asserting on the shape of both properties, and that a separate assertion "we leave out e2e/ct when only one testing type is detected" (which I added below) is the way to go.